spoutbreeze-manager/src/main/java/org/spoutbreeze/spoutbreezemanager/services/BroadcastingAssigner.java

```diff
@@ -21,69 +21,95 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.spoutbreeze.commons.data.broadcast.BroadcastJdbcRepository;
+import org.spoutbreeze.commons.entities.Agent;
 import org.spoutbreeze.commons.entities.Broadcast;
+import org.spoutbreeze.spoutbreezemanager.models.BroadcastMessage;
+import org.spoutbreeze.spoutbreezemanager.repository.BroadcastRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
 
+import javax.transaction.Transactional;
+import java.util.Optional;
+
+@Component
 public class BroadcastingAssigner {
+//
+//    private static final Logger logger = LoggerFactory.getLogger(BroadcastingAssigner.class);
+//
+//    private Boolean isAssigning = false;
+//
+//    @Autowired
+//    @Qualifier("agentsService")
+//    private AgentsService agentsManager;
+//
+//    @Autowired
+//    @Qualifier("assignmentJdbcRepository")
+//    private BroadcastJdbcRepository broadcastJdbcRepository;
+//
+//    // Check if there is any broadcast to assign to an agent then create
+//    // @todo: replace @Scheduled task with real-time message (redis)
+//    @Scheduled(fixedRate = 2500)
+//    public void scheduleAssignments() {
+//        try {
+//            assignRecordingsToAgents();
+//            endAssignement();
+//        } catch (InterruptedException e) {
+//            logger.error("CRITICAL !! An excpeted error occured during assigning the recordings.", e);
+//        }
+//    }
+//
+//    /**
+//     * Fetches the recordings and distribute them to agents
+//     *
+//     * @throws InterruptedException
+//     */
+//    private void assignRecordingsToAgents() throws InterruptedException {
+//        // Do nothing if an assignment is being made
+//        if (assignementInProgess())
+//            return;
+//
+//        // Create a lock to inform the other services that an assignment is in progress

```

Comment:
As far as I know, JPA gives transactional level locking. So do we really need this locking mechanism when a new entry is created? 

We can write a multithreaded test case and check both approaches.